### PR TITLE
Expand comment on managedIdentityResourceId

### PR DESCRIPTION
**Pull request by vplauzon**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by vplauzon** [comment]
1. Change a property description (no change in the swagger structure per se)
2. This is done to update the internal documentation (https://docs.microsoft.com/en-us/azure/templates/microsoft.kusto/clusters/databases/dataconnections?tabs=bicep#eventgridconnectionproperties) ASAP
3. See previous
4. No SDK refresh required as it is only property description

Adding @orhasban for review.

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2022-02-01/kusto.json [hunk 1]
```diff
@@ -4290,7 +4290,7 @@
         },
         "managedIdentityResourceId": {
           "type": "string",
-          "description": "The resource ID of a managed identity (system or user assigned) to be used to authenticate with event hub."
+          "description": "If empty, connection will use Event Hub SAS key.  If Cluser resource ID is provided, the connection will use System Assign Identity.  If User Assigned Identity resource ID is provided, the connection will use that User Assigned Identity."
```
Comment (orhasban) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2022-02-01/kusto.json:4293]: 1. I wouldn't use "Event Hub SAS key" term, but Non MI based Data connection - you don't want to expose the magic.
2. The structure better be as follows:
Leave empty for Non Managed Identity based data connection. For system assigned identity provide cluster resource Id. For user assigned identity (UAI) provide the UAI resource Id.

**Review by orhasban — APPROVED** [review]

**Comment by vplauzon** [comment]
Adding @weidongxu-microsoft to review.

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2022-02-01/kusto.json (hunk 1 above)
Comment (vplauzon) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2022-02-01/kusto.json]: I've implemented suggested changes.

**Review by ruowan — APPROVED** [review]

**Pull request merged**